src/node_crypto.cc

```diff
@@ -923,6 +923,22 @@ void SecureContext::SetECDHCurve(const FunctionCallbackInfo<Value>& args) {
 
   node::Utf8Value curve(env->isolate(), args[0]);
 
+  SSL_CTX_set_options(sc->ctx_, SSL_OP_SINGLE_ECDH_USE);
+
+#if defined(SSL_CTX_set1_curves_list) || defined(SSL_CTRL_SET_CURVES_LIST)
+# if defined(SSL_CTX_set_ecdh_auto)
+  SSL_CTX_set_ecdh_auto(sc->ctx_, 1);
+# endif
+
+  if (strcmp(*curve, "auto") == 0)
+    return;
+
+  if (!SSL_CTX_set1_curves_list(sc->ctx_, *curve))
```

Comment:
One (possibly academic) drawback to `SSL_CTX_set1_curves_list()` is that it won't let you set more than ~30 curves.

I.e., `options = { ecdhCurve: crypto.getCurves().join(':') }` won't work because there are over 80 different curves.

Reply:
Supported curves are listed [here](https://github.com/nodejs/node/blob/8bd1668ca5fe0b0d3f78609ba3d6bc173bce6f98/deps/openssl/openssl/ssl/t1_lib.c#L389). Does the documentation need to mention the lists?

Reply:
Yes, I think that's a good idea.  I was initially going to suggest to just refer readers to the relevant RFCs but on second thought, that would work okay for Brainpool (RFC 7027) but RFC 4492 is too much of a grab bag of different algorithms.